lib/stylesheet/manager.rb

```diff
@@ -211,54 +212,33 @@ def stylesheet_details(target = :desktop, media = 'all')
 
     @@lock.synchronize do
       stylesheets = []
-      stale_theme_ids = []
-      theme_ids = is_theme_target ? @theme_ids : [nil]
-
-      theme_ids.each do |theme_id|
-        cache_key = "path_#{target}_#{theme_id}_#{current_hostname}"
-
-        if href = cache[cache_key]
-          stylesheets << {
-            target: target,
-            theme_id: theme_id,
-            new_href: href
-          }
-        else
-          stale_theme_ids << theme_id
-        end
-      end
-
-      scss_checker = ScssChecker.new(target, stale_theme_ids)
 
       if is_theme_target
-        themes = load_themes(stale_theme_ids)
-
+        scss_checker = ScssChecker.new(target, @theme_ids)
+        themes = load_themes(@theme_ids)
         themes.each do |theme|
           theme_id = theme&.id
-          data = { target: target, theme_id: theme_id }
+          data = { target: target, theme_id: theme_id, theme_name: theme&.name.downcase, remote: theme.remote_theme_id? }
           builder = Builder.new(target: target, theme: theme, manager: self)
-          is_theme = builder.is_theme?
-          has_theme = builder.theme.present?
-
-          if is_theme && !has_theme
-            next
-          else
-            next if is_theme && builder.theme&.component && !scss_checker.has_scss(theme_id)
-            builder.compile unless File.exists?(builder.stylesheet_fullpath)
-            href = builder.stylesheet_path(current_hostname)
-            cache.defer_set("path_#{target}_#{theme_id}_#{current_hostname}", href)
-          end
+
+          next if builder.theme&.component && !scss_checker.has_scss(theme_id)
+          builder.compile unless File.exists?(builder.stylesheet_fullpath)
+          href = builder.stylesheet_path(current_hostname)
 
           data[:new_href] = href
           stylesheets << data
         end
+
+        if stylesheets.size > 1
+          stylesheets = stylesheets.sort_by { |s|
+            [s[:remote] ? 0 : 1, s[:theme_id] == @theme_id ? 1 : 0, s[:theme_name]]
+          }
```

Comment:
```suggestion
          stylesheets = stylesheets.sort_by do |s|
            [
              s[:remote] ? 0 : 1, 
              s[:theme_id] == @theme_id ? 1 : 0, 
              s[:theme_name]
            ]
          end
```

Minor but I find this easier to read as compared to all the code being in a single line.